### PR TITLE
Preserve published model scientific baselines

### LINKED ISSUE #87: Add scientific fixtures and a supported-model matrix
## Outcome

Establish the scientific compatibility contract before deleting or reorganizing historical sources.

## Acceptance criteria

- Add a machine-readable support matrix for GPN, GPN-Star, deprecated GPN-MSA inference, Sorghum gene-expression inference, and PhyloGPN inference.
- Pin immutable Hugging Face revisions for the representative published checkpoints.
- Add small, non-pickle fixtures with biological coordinates, species/allele order, model revision, dtype/device, expected logits/probabilities/LLRs or expression outputs, and explicit tolerances.
- Derive MSA fixtures from existing shared SCF alignments without downloading or committing a whole-genome MSA.
- Reproduce and independently review the notebook-derived expected values.
- Run fixture regressions offline on every PR; keep network/model downloads outside normal CI.
- Document score sign, masking, coordinate, soft-mask, and calibration conventions.

This issue does not include architecture refactors; behavior is captured first.

### PR DESCRIPTION
**Pull request by gonzalobenegas**
## Summary

- preserve pinned likelihood/logit baselines for GPN, GPN-MSA, GPN-Star,
  PhyloGPN, and the Sorghum expression model
- add small model-agnostic scoring helpers with explicit alt-minus-ref semantics
- fix Transformers 5 loading of the published GPN-MSA decoder bias
- replace PhyloGPN import-time downloads and remote-code dependence with an
  explicitly registered local AutoClass implementation
- add an offline Transformers 4.46 lower-bound CI job; ordinary CI remains
  network-free and there is no scheduled Hugging Face check

## Scientific provenance

- vendors only a 3.5 KiB, 128 bp MSA fixture; no whole MSA was downloaded
- records immutable model, dataset, and calibration revisions and hashes
- includes a deterministic utility that regenerates the tiny fixture from an
  explicitly supplied local Zarr and verifies both the raw slice and final NPZ
- verifies the complete GPN-Star-to-GPN-MSA species subset relationship
- binds Sorghum output labels and the example sequence/coordinates to the pinned
  dataset artifacts
- verifies raw and calibrated GPN-Star LLRs

The checked-in PhyloGPN notebook value is stale relative to the pinned current
weights. Both the pinned remote implementation under Transformers 4.48.3 and the
new local implementation produce C-to-T LLR 0.7954897; the notebook will be
refreshed in the documentation layer.

## Validation

- 86 offline tests passed; 5 live tests intentionally skipped
- all 5 pinned published-model integration tests passed once
- 18 focused tests passed on Python 3.11 / Transformers 4.46.3
- wheel and source distribution build, Twine checks, and wheel-content checks
  passed
- independent scientific and engineering reviews completed, followed by closure
  reviews with no remaining findings

Closes #87

### CODE REVIEW

File: gpn/_registration.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
 from threading import Lock
 from typing import Literal
 
-ModelFamily = Literal["gpn", "star"]
+ModelFamily = Literal["gpn", "phylogpn", "star"]
 
-_FAMILIES: tuple[ModelFamily, ...] = ("gpn", "star")
+_FAMILIES: tuple[ModelFamily, ...] = ("gpn", "phylogpn", "star")
```
Comment (gonzalobenegas) [inline gpn/_registration.py]: could you clarify why GPN-MSA is not here? (not saying it should necessarily be)

File: gpn/_registration.py [hunk 2]
```diff
@@ -94,3 +96,16 @@ def _register_star() -> None:
...
+    from transformers import AutoConfig, AutoModel, AutoTokenizer
+
+    from .phylogpn import PhyloGPNConfig, PhyloGPNModel, PhyloGPNTokenizer
+
+    AutoConfig.register("phylogpn", PhyloGPNConfig)
+    AutoModel.register(PhyloGPNConfig, PhyloGPNModel)
+    AutoTokenizer.register(
+        PhyloGPNConfig,
+        slow_tokenizer_class=PhyloGPNTokenizer,
+    )
```
Comment (gonzalobenegas) [inline gpn/_registration.py:114]: why is phylogpn different from the rest?

File: gpn/phylo/model.py [hunk 3]
```diff
@@ -1,5 +1,411 @@
-from transformers import AutoModel, AutoTokenizer
+"""Maintained PhyloGPN configuration, tokenizer, and inference model."""
```
Comment (gonzalobenegas) [inline gpn/phylo/model.py:1]: Why is this as a string? I'd like to discuss options here.

File: gpn/scoring.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
+"""Small, model-agnostic helpers for nucleotide likelihood scores."""
+
+from __future__ import annotations
```
Comment (gonzalobenegas) [inline gpn/scoring.py]: I'm not sure we need this given our python versions?

Comment (gonzalobenegas) [reply gpn/scoring.py]: maybe we could have this in AGENTS.md "Use Python 3.11+ syntax (list[str], X | None); reach for typing only for constructs that still require it.
"

File: gpn/_registration.py (hunk 1 above)
Comment (gonzalobenegas) [reply gpn/_registration.py]: Resolved in dc9e6e7. GPN-MSA now has its own explicit msa selector and model implementation in gpn/msa/model.py. register_auto_classes("msa") registers only the published GPN-MSA checkpoint classes, and registration is tested not to import SS.

File: gpn/_registration.py (hunk 2 above)
Comment (gonzalobenegas) [reply gpn/_registration.py:114]: Resolved in dc9e6e7. PhyloGPN now follows the same public shape as the other families: register_auto_classes("phylo") and gpn/phylo/model.py. The serialized model_type remains phylogpn only because published checkpoint configs already contain that value.

File: gpn/phylo/model.py (hunk 3 above)
Comment (gonzalobenegas) [reply gpn/phylo/model.py:1]: This line is the normal Python module docstring, so it is intentionally a string literal rather than executable configuration. The implementation is now colocated in gpn/phylo/model.py; registration uses the explicit phylo selector.

File: gpn/scoring.py (hunk 4 above)
Comment (gonzalobenegas) [reply gpn/scoring.py]: Resolved. The postponed-annotations import was removed from scoring, and AGENTS.md now documents Python 3.11+ syntax plus the rule to use postponed annotations only when runtime forward references or cycles actually require it. The final audit removed every gratuitous remaining instance.

File: gpn/phylo/model.py [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+
+from __future__ import annotations
```
Comment (gonzalobenegas) [inline gpn/phylo/model.py:3]: I thought we have gotten rid of these?

Comment (gonzalobenegas) [reply gpn/phylo/model.py:3]: didn't we mention putting guidance in AGENTS.md? When will we do that?

Comment (gonzalobenegas) [reply gpn/phylo/model.py:3]: Yes. AGENTS.md is introduced in PR #94, so that is the owning layer. It already contains the Python rule we discussed: use Python 3.11+ syntax such as list[str] and X | None, reach for typing only when still required, and add future annotations only for a real forward-reference/import-cycle/runtime need. I also added ce86be7 there to make the architectural rule explicit: family-specific code stays under ss/msa/star/phylo, top-level modules must be shared by at least two families, and reuse is by composition rather than inheritance.
